packages/ui/src/utilities/parseSearchParams.ts

```diff
@@ -0,0 +1,12 @@
+import type { ReadonlyURLSearchParams } from 'next/navigation.js'
+
+import * as qs from 'qs-esm'
+
+export function parseSearchParams(params: ReadonlyURLSearchParams): qs.ParsedQs {
```

Comment:
Should the argument just be a string? If you called this function multiple times in the same file this would be re-parsing the object every time. I think this function is still handy because it provides a single point for how parsing params should be done as opposed to every file using qs directing and needing to know what options to pass.

Reply:
>  If this function was called with the same object multiple times it would be running toString for no reason

You simply wouldn't call the function multiple times in your component, and ensure it only runs when you need it. Passing the raw url search params directly from the `next/navigation` is nice bc there are no extra steps are needed.

Reply:
It looks like NextJS provides a stable object reference when you use `useSearchParams` so this should be fine. My concern was that this would take away the ability to memoize the result. But I think it should work.